controllers/webhook/ksvc_validator.go

```diff
@@ -0,0 +1,113 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+	types2 "k8s.io/apimachinery/pkg/types"
+	knservingv1 "knative.dev/serving/pkg/apis/serving/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/opendatahub-io/odh-model-controller/controllers/constants"
+	"github.com/opendatahub-io/odh-model-controller/controllers/resources"
+)
+
+// +kubebuilder:webhook:admissionReviewVersions=v1,path=/validate-serving-knative-dev-v1-service,mutating=false,failurePolicy=fail,groups="serving.knative.dev",resources=services,verbs=create,versions=v1,name=validating.ksvc.odh-model-controller.opendatahub.io,sideEffects=None
+
+type ksvcValidator struct {
+	client client.Client
+}
+
+func NewKsvcValidator(client client.Client) *ksvcValidator {
+	return &ksvcValidator{client: client}
+}
+
+func (v *ksvcValidator) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	log := logf.FromContext(ctx).WithName("KsvcValidatingWebhook")
+	ksvc, ok := obj.(*knservingv1.Service)
+	if !ok {
+		return fmt.Errorf("expected a Knative Service but got a %T", obj)
+	}
+
+	log = log.WithValues("namespace", ksvc.Namespace, "ksvc", ksvc.Name)
+	log.Info("Validating Knative Service")
+
+	// If there is an explicit intent for not having a sidecar, skip validation
+	ksvcTemplateMeta := ksvc.Spec.Template.GetObjectMeta()
+	if ksvcTemplateMeta != nil {
+		if templateAnnotations := ksvcTemplateMeta.GetAnnotations(); templateAnnotations[constants.IstioSidecarInjectAnnotationName] == "false" {
+			log.V(1).Info("Skipping validation of Knative Service because there is an explicit intent to exclude it from the Service Mesh")
+			return nil
+		}
+	}
+
+	// Only validate the KSVC if it is owned by KServe controller
+	ksvcMetadata := ksvc.GetObjectMeta()
+	if ksvcMetadata == nil {
+		log.V(1).Info("Skipping validation of Knative Service because it does not have metadata")
+		return nil
+	}
+	ksvcOwnerReferences := ksvcMetadata.GetOwnerReferences()
```

Comment:
TBH, I don't know.

Reply:
seems that, it comes from ksvc, ok := obj.(*knservingv1.Service)
Can this bring everything that is present in the cluster?

Reply:
That would be weird, IMO. At least for KServe case, I expect a single entry.
If you are using Knative directly, there would be no entries at all.

Users or other controllers can change it. It definitely can grew out of proportion, but still it would be intentional.

See [kubernetes docs](https://kubernetes.io/docs/concepts/overview/working-with-objects/owners-dependents/9) fore more info.